docs/src/components/SchemaDocs.astro

```diff
@@ -0,0 +1,142 @@
+---
+/**
+ * Render parts of the JSON schema into a table, given a `group` to render.
+ * Properties in the schema can be assigned to groups by adding a new "groups": ["group1", "group2"] key to them.
+ * This way, you can split schema definition into semantic groups.
+ */
+import { type JSONSchema7, type JSONSchema7Definition, type JSONSchema7TypeName, type JSONSchema7Type } from 'json-schema';
+import MarkdownRenderer from './MarkdownRenderer.astro';
+import rawSchema from '../values.schema.json';
+
+// the 'as any' isn't pretty but needed because our schema doesn't actually conform to the type.
+// it's still nice to use the type to get at least some type hinting when developing.
+let schema: JSONSchema7 = rawSchema as any;
```

Comment:
What are the differences? Would it be easy to extend/adapt the type to fit?

Reply:
Hmm, since it's all just in this class and just a few lines I didn't think it's worth it - even with the type fixed the code below would still have to check for presence/absence of the fields. The types are recursively defined and it's a bit of a complex structure, so I didn't bother. I think it's definitely not _easy_ (maybe also not _hard_ but I think it's tedious and possibly a lot of lines of code)

Reply:
ok, makes sense!